android/src/main/kotlin/com/example/android_locator_plugin/AndroidLocatorPlugin.kt

```diff
@@ -1,35 +1,192 @@
+                if (events != null) {
+                    accessEventSource = events
+                }
+            }
+
+            override fun onCancel(arguments: Any?) {
+            }
+        }
+
+    override fun onAttachedToEngine(@NonNull flutterPluginBinding: FlutterPlugin.FlutterPluginBinding) {
+        context = flutterPluginBinding.applicationContext
+        channel = MethodChannel(flutterPluginBinding.binaryMessenger, ANDROID_LOCATOR_PLUGIN)
+        channel.setMethodCallHandler(this)
+        accessEventChannel = EventChannel(flutterPluginBinding.binaryMessenger, ACCESS_STREAM)
+        accessEventChannel.setStreamHandler(accessStreamHandler)
+        locationEventChannel = EventChannel(flutterPluginBinding.binaryMessenger, EVENT_LOCATION)
+        locationEventChannel.setStreamHandler(locationStreamHandler)
+    }
+
+    override fun onMethodCall(@NonNull call: MethodCall, @NonNull result: Result) {
+        when (call.method) {
+            INITIALIZE -> initializePlugin(result)
+            CHECK_PERMISSION -> checkPermission(result)
+            REQUEST_PERMISSION -> requestPermission(result)
+            else -> result.notImplemented()
+        }
+    }
+
+    private fun requestPermission(result: Result) {
+        context.run {
+            activity.apply {
+                requestPermissions(
+                    this,
+                    arrayOf(
+                        Manifest.permission.ACCESS_COARSE_LOCATION,
+                        Manifest.permission.ACCESS_FINE_LOCATION
+                    ),
+                    REQUEST_CODE
+                )
+            }
+        }
+        result.success(null)
+    }
+
+    private fun checkPermission(result: Result) {
+        val granted = permissionGranted()
+        accessEventSource.success(if (granted) ACCESS_GRANTED else ACCESS_DENIED)
+        result.success(granted)
+    }
+
+    private fun initializePlugin(result: Result) {
+        fusedLocationClient = LocationServices.getFusedLocationProviderClient(activity)
+        val stringBuilder = StringBuilder()
+        val permission = permissionGranted()
+        accessEventSource.success(if (permission) PLUGIN_INITIALIZED else PLUGIN_NEED_PERMISSION)
+        if (permission) {
+            locationRequest = LocationRequest.create()
+            with(locationRequest) {
+                interval = LOCATION_REQUEST_INTERVAL
+                priority = PRIORITY_HIGH_ACCURACY
+            }
+            locationCallback = object : LocationCallback() {
+                override fun onLocationResult(locationResult: LocationResult?) {
+                    locationEventSource.success(
+                        "${stringBuilder.append(LONGITUDE)} ${locationResult?.lastLocation?.longitude.toString()} ${
+                            stringBuilder.append(
+                                LATITUDE
+                            )
+                        } ${locationResult?.lastLocation?.latitude.toString()}"
```

Comment:
Instead to do this, you can create StreamBuilder without using a variable, so this can be like this:
StringBuilder().append(LONGITUDE).append(...)....append().toString()

Reply:
The method that involves this approach was removed momentary. It would be applied in the next functionality.